src/main/java/org/bitcoinj/core/AuxPoW.java

```diff
@@ -423,12 +423,13 @@ public Transaction getTransaction() {
      * @return true if the shorter array is present at the offset, false otherwise.
      */
     static boolean arrayMatch(byte[] script, int offset, byte[] subArray) {
-        for (int matchIdx = 0; matchIdx + offset < script.length && matchIdx < subArray.length; matchIdx++) {
+        int matchIdx;
```

Comment:
This is worth getting in quickly even while pondering the rest, could you make this and its corresponding test a separate PR I can merge ASAP please?